chalicelib/lib/helpers.py

```diff
@@ -25,17 +28,64 @@ def date_range(start_date, stop_date):
 
 
 def validate_date(date, format_str) -> bool:
+    """
+    Validate a string in date format is valid
+    """
 
     try:
-        if ":" in date:
-            start, stop = date.split(":")
-            datetime.strptime(start, format_str)
-            datetime.strptime(stop, format_str)
-            if start > stop:
-                return False
-        else:
-            datetime.strptime(date, format_str)
-    except TypeError:
+        datetime.strptime(date, format_str)
+    except (TypeError, ValueError) as error:
+        log.error(f"Unable to validate date {date}. Error was: {error}")
         return False
 
     return True
+
+
+def parse_date(date: str, format_str: str = "%Y-%m-%d") -> dict:
+    """
+    Parse the date from a string.
+    Expects these formats:
+    "today"
+    "2019-01-01"
+    "2019-01-01:2019-01-02"
+    """
+
+    dates = dict()
+    if date == "today":
+        dates["today"] = datetime.now()
+
+    if ":" in date:
+        first_date, second_date = date.split(":")
```

Comment:
date = "1:2:3" would lead to ValueError.  Consider using date.split(":")[:2] here or wrap in try except